### PR TITLE
Show download progress

### PR DESCRIPTION
**Pull request by peregrineshahin**

### CODE REVIEW

**Comment by ppigazzini** [comment]
```yaml
Apr 30 16:37:53 dfts-0 pserve[6108]:   File "/home/usr00/fishtest/server/fishtest/templates/base.mak", line 570, in render_body
Apr 30 16:37:53 dfts-0 pserve[6108]:     <div>${self.body()}</div>
Apr 30 16:37:53 dfts-0 pserve[6108]:   File "/home/usr00/fishtest/server/fishtest/templates/tests_view.mak", line 557, in render_body
Apr 30 16:37:53 dfts-0 pserve[6108]:     button.textContent = `Downloading.. ${progressFormatted} / ${contentLengthFormatted}`;
Apr 30 16:37:53 dfts-0 pserve[6108]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/mako/runtime.py", line 230, in __str__
Apr 30 16:37:53 dfts-0 pserve[6108]:     raise NameError("Undefined")
Apr 30 16:37:53 dfts-0 pserve[6108]: NameError: Undefined
```

File: server/fishtest/static/js/application.js [hunk 1]
```diff
@@ -66,6 +66,20 @@ function getCookie(cookieName) {
     .find(([name]) => name === cookieName)?.[1];
 }
 
+function formatBytes(bytes) {
```
Comment (ppigazzini) [inline server/fishtest/static/js/application.js:69]: this should be equivalent:
```js
function formatBytes(bytes) {
  const units = ['B', 'KiB', 'MiB', 'GiB', 'TiB'];
  let unitIndex = 0;
  while (bytes >= 1024 && unitIndex < units.length - 1) {
    bytes /= 1024;
    unitIndex++;
  }
  return `${bytes.toFixed(2)} ${units[unitIndex]}`;
}
```

**Comment by peregrineshahin** [comment]
I keep forgetting that Mako disallows the javascript template literals.. could you please fix with correct syntax... afk

**Comment by peregrineshahin** [comment]
Nevermind done.

**Comment by ppigazzini** [comment]
DEV already works with the fix.

```diff
diff --git a/server/fishtest/templates/tests_view.mak b/server/fishtest/templates/tests_view.mak
index 4ee112e..76afe58 100644
--- a/server/fishtest/templates/tests_view.mak
+++ b/server/fishtest/templates/tests_view.mak
@@ -554,7 +554,7 @@
         receivedLength += value.length;

         const progressFormatted = formatBytes(receivedLength);
-        button.textContent = `Downloading.. ${progressFormatted} / ${contentLengthFormatted}`;
+       button.textContent = "Downloading... " + progressFormatted + " / " + contentLengthFormatted;
       }

       // Combine all chunks into a single Blob
```

**Comment by peregrineshahin** [comment]
LOL Yes, this is exaclty what I usually use/used to fix this delima.

File: server/fishtest/templates/tests_view.mak [hunk 2]
```diff
@@ -530,7 +537,28 @@
...
+
+        if (done) {
+          break;
+        }
+
+        chunks.push(value);
+        receivedLength += value.length;
+
+        const progressFormatted = formatBytes(receivedLength);
+        button.textContent = "Downloading... " + progressFormatted + " / " + contentLengthFormatted;
```
Comment (ppigazzini) [inline server/fishtest/templates/tests_view.mak:557]: The text is too wide for the button with the browser half the screen.
Dropping `"Downloading... " +` ?

Comment (peregrineshahin) [reply server/fishtest/templates/tests_view.mak]: We should add a maximum width also.

**Review by ppigazzini — APPROVED** [review]
Looks good on PROD

image

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin

**Pull request merged**